App/src/main/java/com/gluonhq/chat/views/HomePresenter.java

```diff
@@ -49,4 +60,27 @@ private void changeOrientation(double width) {
                     .ifPresent(p -> ((PortraitPresenter) p).loadChat());
         }
     }
+
+    private void showProgressIndicator() {
+        final ObservableList<Channel> channels = service.getChannels();
+        if (channels.isEmpty()) {
+            final VBox vBox = new VBox(10, new ProgressIndicator(), new Label("Retrieving contacts.."));
+            vBox.setAlignment(Pos.CENTER);
+            homeView.setCenter(new StackPane(vBox));
+            channels.addListener(new InvalidationListener() {
+                @Override
+                public void invalidated(Observable o) {
+                    if (homeView.isShowing()) {
+                        changeOrientation(homeView.getScene().getWidth());
+                    }
+                    channels.removeListener(this);
+                }
+            });
+        } else {
+            if (homeView.isShowing()) {
+                changeOrientation(homeView.getScene().getWidth());
+            }
+        }
+        service.initializeService();
```

Comment:
I don't think this belongs here. 
The service should only be initialized once, and this is done in the ChannelPresenter. By moving it here, it will never get called when `service.loggedUser() == null` (which is when we are running for the first time).
What happens if you remove it here, but re-introduce it in ChannelPresenter, line 65?

Reply:
I got now that service.loggedUser is always null when we didn't initialize the service yet, so this is fine.